checks/evaluation/branch_protection_test.go

```diff
@@ -17,564 +17,1396 @@ package evaluation
 import (
 	"testing"
 
-	"github.com/ossf/scorecard/v4/checker"
-	"github.com/ossf/scorecard/v4/clients"
+	"github.com/ossf/scorecard/v4/finding"
 	scut "github.com/ossf/scorecard/v4/utests"
 )
 
-func testScore(branch *clients.BranchRef, codeownersFiles []string, dl checker.DetailLogger) (int, error) {
-	var score levelScore
-	score.scores.basic, score.maxes.basic = basicNonAdminProtection(branch, dl)
-	score.scores.review, score.maxes.review = nonAdminReviewProtection(branch)
-	score.scores.adminReview, score.maxes.adminReview = adminReviewProtection(branch, dl)
-	score.scores.context, score.maxes.context = nonAdminContextProtection(branch, dl)
-	score.scores.thoroughReview, score.maxes.thoroughReview = nonAdminThoroughReviewProtection(branch, dl)
-	score.scores.adminThoroughReview, score.maxes.adminThoroughReview = adminThoroughReviewProtection(branch, dl)
-	score.scores.codeownerReview, score.maxes.codeownerReview = codeownerBranchProtection(branch, codeownersFiles, dl)
-
-	return computeFinalScore([]levelScore{score})
-}
-
-// TODO: order of tests to have progressive scores.
-func TestIsBranchProtected(t *testing.T) {
+func TestBranchProtection(t *testing.T) {
 	t.Parallel()
-	trueVal := true
-	falseVal := false
-	var zeroVal int32
-	var oneVal int32 = 1
-	branchVal := "branch-name"
 	tests := []struct {
-		name            string
-		branch          *clients.BranchRef
-		codeownersFiles []string
-		expected        scut.TestReturn
+		name     string
+		findings []finding.Finding
+		result   scut.TestReturn
 	}{
 		{
-			name: "GitHub default settings",
```

Comment:
To a large extent, `&falseVal` has been changed to `OutcomeNegative` and `&trueVal` has been changed to `OutcomePositive`. Missing values or `nil` can be either `OutcomeNegative` or `OutcomeNotAvailable` depending on the probe.

`RequireLinearHistory` existed in the test but is not used in the evaluation.

Reply:
I have not changed results from the tests to make a point out of compatibility. 

I have not added new tests, nor have I removed any.